openassessment/xblock/static/js/src/studio/oa_edit_settings.js

```diff
@@ -27,11 +28,38 @@ OpenAssessment.EditSettingsView = function(element, assessmentViews) {
         "#openassessment_submission_due_time"
     ).install();
 
+    new OpenAssessment.SelectControl(
+        $("#openassessment_submission_upload_selector", this.element),
+        {'custom': $("#openassessment_submission_white_listed_file_types_wrapper", this.element)},
+        new OpenAssessment.Notifier([
+            new OpenAssessment.AssessmentToggleListener()
+        ])
+    ).install();
+
     this.leaderboardIntField = new OpenAssessment.IntField(
         $("#openassessment_leaderboard_editor", this.element),
         { min: 0, max: 100 }
     );
 
+    this.fileTypeWhiteListInputField = new OpenAssessment.InputControl(
+        $("#openassessment_submission_white_listed_file_types", this.element),
+        function(value) {
+            var badExts = [];
+            var errors = [];
+            var whiteList = $.map(value.split(','), $.trim);
+            $.each(whiteList, function(index, ext) {
+                if (data.FILE_EXT_BLACK_LIST.indexOf(ext) !== -1) {
+                    badExts.push(ext);
+                }
+            });
+            if (badExts.length) {
+                errors.push(gettext('The following file types are not allowed: ') + badExts.join(','));
+            }
```

Comment:
Note that I am still able to enter "DMG" or "EXE" and save without error due to the blacklist. (see uppercase_extensions_no_error.png, where I edited a released unit)
When I edit the ORA component again, the uppercase extensions have been converted to lowercase. (see uppercase_after_save.png)
When I save this time, I get the error message. (see lowercase_extensions_error.png)

![lowercase_extensions_error](https://cloud.githubusercontent.com/assets/6392274/10226317/b0293c84-6834-11e5-9986-24a7408d433d.png)
![uppercase_after_save](https://cloud.githubusercontent.com/assets/6392274/10226318/b0297cd0-6834-11e5-8c6b-97f0bb03205e.png)
![uppercase_extensions_no_error](https://cloud.githubusercontent.com/assets/6392274/10226319/b02be0ec-6834-11e5-9ee7-739082e884b2.png)